### PR TITLE
putting "Improvements to charging" back in place, but without the EBC default

### PR DESCRIPTION
**Pull request by marc-hanheide**
Reverts strands-project/scitos_apps#123

### CODE REVIEW

File: scitos_docking/include/scitos_docking/CChargingState.h [hunk 1]
```diff
@@ -70,6 +73,9 @@ typedef enum{
 	STATE_TEST_MOVE,
 	STATE_INIT,
 	STATE_UNDOCK_INIT,
+	STATE_CALIBRATE_INIT,
+	STATE_HEAD_ON,
```
Comment (lucasb-eyer) [inline scitos_docking/include/scitos_docking/CChargingState.h:77]: I think these two should be the other way around, i.e. `OFF` first, then `ON`.

**Comment by lucasb-eyer** [comment]
Besides my inline comment, let me point out that the name `/head_xtion` is hardcoded. If we go that way, we could also take the parametrization of the name out of all of our launchfiles, that'll make them simpler.....

The 24V EBC name being hardcoded doesn't help flexibility either, but it doesn't look like this code is intended for the wider public anyways, given all the hardcoding.

**Comment by marc-hanheide** [comment]
I think, yes, it has a number of hard-coded assumptions in it. But fixing all of these will be a bigger job. For your line comment, is that a style question, or an actual problem?

**Comment by lucasb-eyer** [comment]
Well if I understand the code correctly, it will tell you "turning head off" when it's actually turning it on and vice-versa, so it might make someone trying to understand what happens from the output crazy.

**Comment by marc-hanheide** [comment]
Ah, now I see! Yes, the problem is that this only showed an excerpt of the code in here, so I didn't see this...

File: scitos_docking/include/scitos_docking/CChargingState.h (hunk 1 above)
Comment (marc-hanheide) [inline scitos_docking/include/scitos_docking/CChargingState.h:77]: fixed in this PR

**Comment by marc-hanheide** [comment]
fixed it (but changed the order of the text, not the states)

**Comment by marc-hanheide** [comment]
(reasonably) happy with it now?

**Comment by lucasb-eyer** [comment]
That's all I saw during a quick glance, I cannot judge the rest, but really, @gestom should be the one checking this...

**Comment by gestom** [comment]
Before implementing the visual charging, I have pointed out that (obviously) it will not work in darkness.
During the GA1, BHAM complained that the robot cannot dock in a dark room and we have proposed four solutions. As you can see, I have actually preferred to simply keep the charging area lit: https://github.com/strands-project/scitos_apps/issues/40.

However, the project coordinator decided to solve the problem by asking us to add the light source to the robot, see "https://github.com/strands-project/scitos_apps/issues/40" claiming that it will be useful for the security scenario. The coordinator also asked the other partners (see 'local decision' https://github.com/strands-project/strands_management/issues/19) to decide about the solution, where RTWH said that they will not use the onboard light. As a summary, BHAM and UoL indicated the onboard light use and RTWH not. Everybody was OK with this for more than one year until RTWH started to use the EBC port to power the onboard network switch, which caused them to run into problems, because the docking behaviour is switching the port on and off to change the light.

Now, instead of simply powering the switch from another of 12V power source (there are plenty of these on the robot's actual ). Plugging the cable to any other power source would have solve the issue in 30 minutes. Instead of it, RTWH decided to change the docking behaviour code, breaking the charging functionality for UoL.

I have spend enough time with this.

**Comment by lucasb-eyer** [comment]
AFAIK, that specific port was actually planned to be used for powering the mantas on the head, that's why the cable goes up there. So had we switched the switch's EBC, we'd have gotten the same problem as soon as we'd mount the mantas. Since we're no robotics lab, we don't have the tools necessary to do that change in 30min. It was faster to make it configurable in the software.

I pinged you in my PR where I _made it configurable_, and you agreed with it, so I wouldn't consider it breaking. While debugging it, I was surprised to see strands-specific decisions hardcoded in a _scitos_-package, but that's another story.

My comment in #119 was joking, maybe it didn't come across that way. In case I was rude, I apologise, I didn't mean to be, but the burn-out of the past weeks is starting to be noticeable; sorry for that :(

**Pull request merged**